com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceCreator.java

```diff
@@ -36,36 +29,15 @@ ServiceCreator withErrorHandlerSupplier(Supplier<CustomControllerClientErrorHand
         return this;
     }
 
-    public MethodExecution<String> createService(CloudControllerClient client, CloudServiceExtended service, String spaceId) {
+    public MethodExecution<String> createService(CloudControllerClient client, CloudServiceExtended service) {
         return errorHandlerSupplier.get()
-                                   .handleErrorsOrReturnResult(() -> attemptToCreateService(client, service, spaceId));
+                                   .handleErrorsOrReturnResult(() -> attemptToCreateService(client, service));
     }
 
-    private MethodExecution<String> attemptToCreateService(CloudControllerClient client, CloudServiceExtended service, String spaceId) {
+    private MethodExecution<String> attemptToCreateService(CloudControllerClient client, CloudServiceExtended service) {
         assertServiceAttributes(service);
-
-        RestTemplate restTemplate = getRestTemplate(client);
-        String cloudControllerUrl = client.getCloudControllerUrl()
-                                          .toString();
-        CloudServicePlan cloudServicePlan = findPlanForService(client, service);
-
-        Map<String, Object> serviceRequest = createServiceRequest(service, spaceId, cloudServicePlan);
-        String url = getUrl(cloudControllerUrl, CREATE_SERVICE_URL_ACCEPTS_INCOMPLETE_TRUE);
-        HttpEntity<Map<String, Object>> requestEntity = new HttpEntity<>(serviceRequest);
-        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, requestEntity, String.class);
-        return MethodExecution.fromResponseEntity(response);
-    }
-
-    private Map<String, Object> createServiceRequest(CloudServiceExtended service, String spaceId, CloudServicePlan cloudServicePlan) {
-        Map<String, Object> serviceRequest = new HashMap<>();
-        serviceRequest.put(SPACE_GUID, spaceId);
-        serviceRequest.put(SERVICE_NAME, service.getName());
-        serviceRequest.put(SERVICE_PLAN_GUID, cloudServicePlan.getMetadata()
-                                                              .getGuid()
-                                                              .toString());
-        serviceRequest.put(SERVICE_PARAMETERS, service.getCredentials());
-        serviceRequest.put(SERVICE_TAGS, service.getTags());
-        return serviceRequest;
+        client.createService(service);
```

Comment:
Please make sure that this request covers all the Request parameters + calls the correct URL(accepts_incomplete query parameter).

Reply:
If you are sure that this call is enough and is doing all the necessarry putting of request parameters, remove the class and replace every usage of ServiceCreator with `client.createService(service);`